src/ValueFormatters/QuantityHtmlFormatter.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace ValueFormatters;
+
+/**
+ * HTML formatter for quantity values.
+ *
+ * @since 0.5.1
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class QuantityHtmlFormatter extends QuantityFormatter {
+
+	/**
+	 * @param string $number
+	 * @param string $unit
+	 *
+	 * @return string HTML
+	 */
+	protected function applyUnit( $number, $unit ) {
+		$html = htmlspecialchars( $number );
+
+		if ( $this->vocabularyUriFormatter !== null ) {
+			$unit = $this->vocabularyUriFormatter->format( $unit );
+
+			if ( $unit === null ) {
+				return $html;
+			}
+		}
+
+		return $html . ' <span class="unit">' . htmlspecialchars( $unit ) . '</span>';
```

Comment:
The problem with the generic class name is that it would be present elsewhere in the DOM, which would make it harder to find the "correct" element for the unit.